src/Utils/BlockTemplateUtils.php

```diff
@@ -255,27 +255,72 @@ public static function get_template_paths( $base_directory ) {
-			case 'taxonomy-product_tag':
-				return __( 'Products by Tag', 'woo-gutenberg-products-block' );
-			default:
-				// Replace all hyphens and underscores with spaces.
-				return ucwords( preg_replace( '/[\-_]/', ' ', $template_slug ) );
+	public static function get_block_template_title( $template_slug ) {
+		$plugin_template_types = self::get_plugin_block_template_types();
+		if ( isset( $plugin_template_types[ $template_slug ] ) ) {
+			return $plugin_template_types[ $template_slug ]['title'];
+		} else {
+			// Human friendly title converted from the slug.
+			return ucwords( preg_replace( '/[\-_]/', ' ', $template_slug ) );
 		}
 	}
 
+	/**
+	 * Returns template descriptions.
+	 *
+	 * @param string $template_slug The templates slug (e.g. single-product).
+	 * @return string Template description.
+	 */
+	public static function get_block_template_description( $template_slug ) {
+		$plugin_template_types = self::get_plugin_block_template_types();
+		if ( isset( $plugin_template_types[ $template_slug ] ) ) {
+			return $plugin_template_types[ $template_slug ]['description'];
+		}
+	}
+
+	/**
+	 * Returns a filtered list of plugin template types, containing their
+	 * localized titles and descriptions.
+	 *
+	 * @return array The plugin template types.
+	 */
+	public static function get_plugin_block_template_types() {
+		$plugin_template_types = array(
+			'single-product'         => array(
+				'title'       => _x( 'Single Product', 'Template name', 'woo-gutenberg-products-block' ),
+				'description' => __( 'Template used to display the single product.', 'woo-gutenberg-products-block' ),
+			),
+			'archive-product'        => array(
+				'title'       => _x( 'Product Catalog', 'Template name', 'woo-gutenberg-products-block' ),
+				'description' => __( 'Template used to display products.', 'woo-gutenberg-products-block' ),
+			),
+			'taxonomy-product_cat'   => array(
+				'title'       => _x( 'Products by Category', 'Template name', 'woo-gutenberg-products-block' ),
+				'description' => __( 'Template used to display products by category.', 'woo-gutenberg-products-block' ),
+			),
+			'taxonomy-product_tag'   => array(
+				'title'       => _x( 'Products by Tag', 'Template name', 'woo-gutenberg-products-block' ),
+				'description' => __( 'Template used to display products by tag.', 'woo-gutenberg-products-block' ),
+			),
+			'product-search-results' => array(
+				'title'       => _x( 'Product Search Results', 'Template name', 'woo-gutenberg-products-block' ),
+				'description' => __( 'Template used to display search results for products.', 'woo-gutenberg-products-block' ),
+			),
+		);
+
+		/**
+		 * Filters the list of template types.
+		 *
+		 * @param array $plugin_template_types An array of template types, formatted as [ slug => [ title, description ] ].
+		 */
+		return apply_filters( 'woocommerce_gutenberg_products_block_template_types', $plugin_template_types );
```

Comment:
I'm not sure that it is necessary to add this filter. What is the reason behind this?

Reply:
The core has one, thought we could provide an easy way of filtering it in our plugin too. But if we determine it's unnecessary I can remove it.

Reply:
If core has a filter that serves the same purpose as this, then there's no real need for this one. I think we can remove this filter and people can use cores filter instead.

Reply:
Same principle, but different data: https://github.com/WordPress/WordPress/blob/d9266d5bfd6aed72907414728d83f1053c1c2a7a/wp-includes/block-template-utils.php#L189

The core filter allows for filtering the default templates. This one allows for filtering our custom ones.

Reply:
>The core filter allows for filtering the default templates. This one allows for filtering our custom ones.

Can developers who require our custom template data just ensure they set their filter priority to run _after_ ours so they get our custom template data using the original filter anyway?

Reply:
Oh I see. My bad, I thought we were hooking into Cores filter and then applying our own anyway. I still think we should remove this, we have a separate discussion and ideas for extensibility when it comes to Woo templates.

It will be hard to remove a filter once we have added it because then we will have to support it if other plugins/themes start using it.

Reply:
Right, I see your point here, although since this is following the same exact pattern the core uses, I think it would make sense to mirror that.

I am happy to remove it though :)

Reply:
>I am happy to remove it though :)

I think we should remove until we _know_ there is a need for it and that when that time comes that we are extending this functionality in the correct way.